research-software-directory/RSD-as-a-service
# 593 disallow mention update

# Limited update allowed (rsd_admin only)

Closes #593 

Changes proposed in this pull request:
* Manually created mentions can only be updated by rsd_admin role
* RSD user can create custom mention (using ADD button) but they are NOT allowed to update it because currently this item can be referenced in other software/projects.     
* RSD can delete mention from the software/project. If this mention is not referenced by "others" it will also be removed from RSD database
* The source value for custom mentions is changed from "manual" to "RSD". **Historical values in DB should be updated from manual to RSD.**
* The source value for RSD mentions imported from crossref/datacite shows the original source (not RSD as previously was the case)
* Additional explanation is added to info panel on the page and in the create item modal (see print screens)

How to test:
* I think it is better to start with clean db so do the following:
   *  `docker-compose down --volumes` to remove old
   * `docker-compose build` to build
   * `docker-compose up`
* login to rsd as rsd_user and create 2 software items: Software 1 and Software 2
* Edit the Software 1 and navigate to mentions and add custom mention: Custom mention 1
* Confirm that notification is shown in the Modal about not being able to update after save.
* Confirm there is no edit icon on the mention in the list
* Edit the Software 2, search for "Custom mention 1". It should be found indicating the source "RSD"
* Add this mention to Software 2.
* Confirm that there is no edit button
* Delete this mention from Software 2
* Navigate to Software 1. Delete same mention (Custom mention 1). If you removed the reference from Software 2 the mention will also be deleted from RSD.
* Search for Custom mention 1 in any software/project. It should not return result from RSD.
* Login to RSD as admin and repeat complete process of mention creation and referencing in software 1 and 2. As admin you will be able to update custom mention item without errors even after referencing mention in multiple software/projects.

## Example extended info about editing custom item
![image](https://user-images.githubusercontent.com/9204081/197511176-3554911f-cc30-4d9e-8e41-b8bd4100e265.png)

## Example no edit warning on custom output
![image](https://user-images.githubusercontent.com/9204081/197500202-ef20684a-d881-4a55-aaec-0d96c283a11c.png)

PR Checklist:
